### PR TITLE
feat: show warning in comment when coverage report is too long (v1.12.0)

### PR DESCRIPTION
**Pull request by MishaKav**
## What does this PR do?

Brings the truncation warning feature from #298 (by @mschoettle) into `main`, together with release preparation:

- when the comment exceeds GitHub's 65,536-character limit and the detailed coverage report is dropped, a `> [!WARNING]` notice now appears in the comment itself, linking to the job log with the reduction tips — previously the report silently disappeared
- refactor: `tooLongNotice` now reads `MAX_COMMENT_LENGTH` directly instead of receiving it as a parameter (it's a module-level constant in the same file)
- chore: bump version to 1.12.0 and add the CHANGELOG entry

The missing final size enforcement discussed in #298 is intentionally out of scope and tracked in #299.

## Related Issue

Closes #

## Checklist

- [x] Tested locally
- [x] Ran `npm run all`
- [ ] Updated docs (if needed)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01JpeeSEUqmiQrNJA283Njoq

---
_Generated by [Claude Code](https://claude.ai/code/session_01JpeeSEUqmiQrNJA283Njoq)_

---
## Summary by cubic
Shows a warning in the comment when it exceeds GitHub's 65,536-character limit and the detailed coverage report is dropped, instead of silently disappearing. The `> [!WARNING]` notice links to the job log, which lists ways to reduce comment size.

- Refactors `tooLongNotice` to read `MAX_COMMENT_LENGTH` directly instead of receiving it as a parameter.
- Bumps version to `1.12.0` and adds a CHANGELOG entry.

<sup>Written for commit a199849084f0a95a145c212da6f31b03d7ff3c75. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/MishaKav/pytest-coverage-comment/pull/300?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -485,6 +506,9 @@ const main = async (): Promise<void> => {
   }
 
   finalHtml += html;
+  if (tooLongHtml) {
+    finalHtml += finalHtml.length ? `\n\n${tooLongHtml}` : tooLongHtml;
+  }
```
[the remark this answers is not in the document]
Comment (MishaKav) [reply src/index.ts:511]: This is intentionally out of scope for this PR, per the maintainer's decision: the missing final size enforcement is pre-existing on `main` (the fallback chain and `commentLength()` predate this change), and it's tracked in #299 with a proposed fix (`truncateSummary` on the assembled body before posting). This PR only makes the existing truncation behavior visible; the notice is counted in `commentLength()` for the reduction steps.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,13 @@
 # Changelog of the Pytest Coverage Comment
 
+## [Pytest Coverage Comment 1.12.0](https://github.com/MishaKav/pytest-coverage-comment/tree/v1.12.0)
+
+**Release Date:** 2026-08-30
+
+#### Changes
```
[the remark this answers is not in the document]
Comment (MishaKav) [reply CHANGELOG.md:7]: Not applying this one: every release entry in this CHANGELOG (back to 1.0.x) uses `#### Changes` under the `##` release heading — it's the file's established format. Changing only the 1.12.0 entry would make it inconsistent with the rest of the file, and restyling the whole changelog is out of scope for this PR.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,13 @@
...
+
+- show a warning inside the comment itself when it's too long (over 65,536 characters) and the detailed coverage report gets dropped — previously the report silently disappeared and the reason was only visible in the job log. The warning links to the job log, which lists the ways to reduce the comment size (#298), thanks to [@mschoettle](https://github.com/mschoettle) for contribution
```
[the remark this answers is not in the document]
Comment (MishaKav) [reply CHANGELOG.md:9]: Keeping as-is: "thanks to … for contribution" is the phrasing already used by the existing entries on `main` (see the 1.11.2 entry), so the new entry deliberately matches the file's established wording.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/index.ts [hunk 4]
```diff
@@ -419,11 +431,13 @@ const main = async (): Promise<void> => {
...
 
   // every part that ends up in the comment body counts toward the limit
+  let tooLongHtml = '';
   const commentLength = (): number =>
     html.length +
     summaryReport.length +
     failedTestsHtml.length +
-    multipleFilesHtml.length;
+    multipleFilesHtml.length +
+    tooLongHtml.length;
```
[the remark this answers is not in the document]
Comment (MishaKav) [reply src/index.ts:440]: Correct observation, but it's part of a pre-existing pattern: `commentLength()` has never counted any of the assembly separators (nor the watermark) — the ones before `summaryReport`, `failedTestsHtml`, and `multipleFilesHtml` are equally uncounted on `main`. The complete fix (final size enforcement on the assembled body before posting, which subsumes all separator accounting) is intentionally deferred to #299 per the maintainer's decision.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (MishaKav) [reply src/index.ts:511]: Duplicate of the finding already raised (and withdrawn) by Qodo above: the missing final size enforcement is pre-existing on `main` and intentionally out of scope for this PR per the maintainer's decision — it's tracked in #299 with a proposed fix (`truncateSummary` on the assembled body before posting).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/index.ts [hunk 5]
```diff
@@ -485,6 +506,9 @@ const main = async (): Promise<void> => {
   }
 
   finalHtml += html;
+  if (tooLongHtml) {
+    finalHtml += finalHtml.length ? `\n\n${tooLongHtml}` : tooLongHtml;
```
[the remark this answers is not in the document]
Comment (MishaKav) [reply src/index.ts:510]: Same finding as Qodo (withdrawn above) and CodeAnt: the missing final size enforcement is pre-existing on `main` and intentionally out of scope for this PR per the maintainer's decision — tracked in #299 with a proposed fix (`truncateSummary` on the assembled body right before posting).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (MishaKav) [reply src/index.ts:440]: Confirmed intended. Counting the notice in `commentLength()` is deliberate: it makes the reduction steps account for what will actually be posted, instead of letting the notice overflow the limit. In the narrow boundary case you describe (~150 chars), trading a sliver of failed-test content for an explanation of *why* content is missing is the point of this PR — before it, content disappeared with no in-comment explanation at all. The failed tests remain fully visible in the job log either way.

---
_Generated by [Claude Code](https://claude.ai/code)_
